### PR TITLE
Enable parallel cypress

### PR DESCRIPTION
**Pull request by tlrobinson**
Partial fix for #12437

Runs should show up here https://dashboard.cypress.io/projects/a394u1/runs

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -586,6 +586,7 @@ jobs:
         type: string
         default: ""
     executor: << parameters.e >>
+    parallelism: 4
```
Comment (camsaul) [inline .circleci/config.yml:589]: I don't think the `parallelism` parameter works for Docker executors? IIRC it's only for `machine` executors

Comment (tlrobinson) [reply .circleci/config.yml:589]: Not sure if it actually works. I don't see 4 machines running it https://app.circleci.com/pipelines/github/metabase/metabase?branch=cypress-parallel

(separately there's another issue with it being confused by two sets of cypress tests)

**Comment by tlrobinson** [comment]
@paulrosenzweig @camsaul do you want to review this?

**Review by paulrosenzweig — APPROVED** [review]

**Comment by tlrobinson** [comment]
Our Cypress Dashboard account has been upgraded so I'm going to merge this.

**Pull request merged**